dynamic config - ADL for toolconfig & state - show-config-modes command

paul-thompson-helix opened the pull request:
Reboot of the dynamic config on top of c2 master.

paul-thompson-helix committed:
ADL updates for dynamic config

* ADL types for config & state
* createDeploy function updated to create empty placeholder dynamicConfigModes

timbod7 reviewed (approved):
This all looks good to me. I think some documentation showing some example of the different models of configuration.

paul-thompson-helix committed:
Add command to echo dynamic config options from toolconfig

Update demo.

show-config-modes:  list dynamic config names and their modes
show-config-modes  cfgname:  show name and modes for a given cfgname

paul-thompson-helix committed:
Tidy up dynamic config printing functions
